.github/workflows/windows.yml

```diff
@@ -119,7 +123,12 @@ jobs:
           }
 
       - name: Upload package
-        uses: actions/upload-artifact@master
+        uses: actions/upload-artifact@v2
         with:
           name: dosbox-staging-windows-${{ matrix.conf.arch }}-${{ env.VERSION }}
           path: dosbox-staging-windows-${{ matrix.conf.arch }}-${{ env.VERSION }}
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: changelog-${{ env.VERSION }}.txt
+          path: changelog-${{ env.VERSION }}.txt
```

Comment:
For the windows CI run, we have two jobs uploading this same-named artifact. 

GitHub [mentions](https://github.com/actions/upload-artifact/blob/master/README.md#uploading-to-the-same-artifact
):
- _"Uploading to the same artifact multiple times in the same workflow can overwrite and append already uploaded files"_ and,
- _"Warning: Be careful when uploading to the same artifact via multiple jobs as artifacts may become corrupted"_

I'm not sure we really need to work.. these really are the same file - so if the second pass clobbers the first pass, then that's fine. But if they're being appended to (or "corrupted") then we don't want that.   

Maybe the changelog could be part of the zip file?  Or if changelog is in our repo (and not generated on the fly as part of the build), then maybe we could link to it directly and avoid using zipped artifacts to share it?

(this is pretty minor though)

Reply:
@kcgen the whole point of adding a changelog is having a separate file, so I could propagate version info via a file name (because I can't look into an artifact using GitHub REST API) - I mentioned it in PR and in commit message - perhaps I should mention it in a comment also?

As for Windows builds and duplicated artifacts: which artifact is duplicated? I don't see it?

Reply:
`changelog-${{ env.VERSION }}.txt` will have the same name for both release passes through the matrix (both 32bit and 64bit).  I think anyway.. :-)

Reply:
@kcgen Ah, you're right! For some reason, I thought you are talking about zip file with executables.

When implementing I just looked at build results like https://github.com/dosbox-staging/dosbox-staging/actions/runs/158192528 and forgot it's actually a job inside matrix.

In all jobs I tried so far it worked but we don't have any guarantee, that zip file won't be corrupted.

Thank you for spotting this! I'll redesign the solution here.